Automattic/jetpack
# actions: Use GitHub's action/setup-node's new pnpm cache support

#### Changes proposed in this Pull Request:

They've [recently added](https://github.blog/changelog/2021-09-07-github-actions-setup-node-supports-dependency-caching-for-projects-with-monorepo-and-pnpm-package-manager/) support for caching pnpm dependencies. Let's take
advantage of it.

Also of note is that this requires pnpm be installed before calling
setup-node.

#### Jetpack product discussion


None

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Check that the action runs seem to be indicating that caching is being set up properly. This may require re-running jobs.
